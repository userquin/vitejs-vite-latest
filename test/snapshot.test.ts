import { expect, test } from 'vitest';
import { testOutsideInlineSnapshot } from './snapshots-outside';

test('object', () => {
  expect({
    this: { is: new Set(['of', 'snapshot']) },
  }).toMatchSnapshot();
});

test('multiline', () => {
  expect(`
  Hello
    World
`).toMatchSnapshot();
});

test('from outside', () => {
  testOutsideInlineSnapshot();
});

test('with big array', () => {
  expect({
    this: { is: new Set(['one', new Array(30).fill({})]) },
  }).toMatchSnapshot();
});

test('with big string', () => {
  expect({
    this: { is: new Set(['one', new Array(30).fill('zoo').join()]) },
  }).toMatchSnapshot();
});

test('throwing', () => {
  expect(() => {
    throw new Error('omega');
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw 'omega';
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    // eslint-disable-next-line no-throw-literal
    throw { error: 'omega' };
  }).toThrowErrorMatchingSnapshot();
});

test('properties snapshot', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
    name: expect.stringContaining('LeBron'),
  });
});

test.fails('properties snapshot fails', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(String),
  });
});
