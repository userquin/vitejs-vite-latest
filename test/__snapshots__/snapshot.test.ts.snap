// Vitest Snapshot v1

exports[`multiline 1`] = `
"
  Hello
    World
"
`;

exports[`object 1`] = `
{
  "this": {
    "is": Set {
      "of",
      "snapshot",
    },
  },
}
`;

exports[`properties snapshot 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": StringContaining "LeBron",
}
`;

exports[`snapshot 1`] = `
{
  "this": {
    "is": Set {
      "of",
      "snapshot",
    },
  },
}
`;

exports[`snapshot with big array 1`] = `
{
  "this": {
    "is": Set {
      "one",
      [
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
      ],
    },
  },
}
`;

exports[`snapshot with big string 1`] = `
{
  "this": {
    "is": Set {
      "one",
      "zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo",
    },
  },
}
`;

exports[`throwing 1`] = `"omega"`;

exports[`throwing 2`] = `"omega"`;

exports[`throwing 3`] = `
{
  "error": "omega",
}
`;

exports[`throwing snapshots 1`] = `"omega"`;

exports[`throwing snapshots 2`] = `"omega"`;

exports[`throwing snapshots 3`] = `
{
  "error": "omega",
}
`;

exports[`with big array 1`] = `
{
  "this": {
    "is": Set {
      "one",
      [
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
        {},
      ],
    },
  },
}
`;

exports[`with big string 1`] = `
{
  "this": {
    "is": Set {
      "one",
      "zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo,zoo",
    },
  },
}
`;
